bootstrap/bootstrap-kubestellar.sh

```diff
@@ -194,67 +248,131 @@ if [ "$folder" == "" ]; then
-    if [[ ! ":$PATH:" == *":$(get_full_path $folder/kcp/bin):"* ]]; then
-        export PATH=$PATH:$(get_full_path $folder/kcp/bin)
+    ensure_folder "$folder/kcp"
+    kcp_folder=$(get_full_path "$folder/kcp")
+    kcp_bin_folder="$kcp_folder/bin"
+    echo "Downloading kcp+plugins $kcp_version $os_type/$arch_type..."
+    kcp_download
+    echo "Installing kcp+plugins into '$kcp_folder'..."
+    kcp_install
+    if [[ ! ":$PATH:" == *":$kcp_bin_folder:"* ]]; then
+        export PATH=$kcp_bin_folder:$PATH
+        echo "Add kcp folder to the PATH: export PATH=\"$kcp_bin_folder:\$PATH\""
+        user_exports="$user_exports"$'\n'"export PATH=\"$kcp_bin_folder:\$PATH\""
     fi
 fi
 
 # Ensure kcp is running
-if [ "$(kcp_running)" == "false" ]; then
+echo "************************************************"
+echo "Ensure kcp is running"
+echo "************************************************"
+if [ "$(kcp_running)" == "true" ]; then
+    echo "kcp process is running already: pid=$(pgrep kcp) ... skip running."
+    if [ "$(kubeconfig_valid)" == "false" ]; then
+        echo "KUBECONFIG environment variable is not set correctly: KUBECONFIG='$KUBECONFIG' ... exiting!"
+        exit 3
+    fi
+    echo "Using 'KUBECONFIG=$KUBECONFIG'"
+    echo "Waiting for kcp to be ready... it may take a while"
+    until $(kcp_ready)
+    do
+        sleep 1
+    done
+    if [ "$(kcp_version)" != "$kcp_version" ]; then
+        echo "kcp running version $(kcp_version) does not match the desired version $kcp_version ... exiting!"
+        exit 4
+    else
+        echo "kcp version $(kcp_version) ... ok"
+    fi
+else
     if [ "$kcp_address" == "" ]; then
-        if [ "$verbose" != "" ]; then
-            echo "Starting kcp..."
-        fi
+        echo "Running kcp... logfile=$PWD/kcp_log.txt"
         kcp start >& kcp_log.txt &
     else
-        if [ "$verbose" != "" ]; then
-            echo "Starting kcp bound to address $kcp_address..."
-        fi
+        echo "Running kcp bound to address $kcp_address... logfile=$PWD/kcp_log.txt"
         kcp start --bind-address $kcp_address >& kcp_log.txt &
     fi
-    export KUBECONFIG="$(pwd)/.kcp/admin.kubeconfig"
+    export KUBECONFIG="$PWD/.kcp/admin.kubeconfig"
+    echo "Waiting for kcp to be ready... it may take a while"
     sleep 10
     until $(kcp_ready)
     do
         sleep 1
     done
     sleep 10
-    echo "Export KUBECONFIG with the command: export KUBECONFIG='$(pwd)/.kcp/admin.kubeconfig'"
+    if [ "$(kcp_version)" != "$KCP_REQUIRED_VERSION" ]; then
+        echo "kcp version $(kcp_version) is not supported, KubeStellar requires kcp $KCP_REQUIRED_VERSION ... exiting!"
+        exit 4
+    else
+        echo "kcp version $(kcp_version) ... ok"
+    fi
+    echo "Export KUBECONFIG environment variable: export KUBECONFIG=\"$KUBECONFIG\""
+    user_exports="$user_exports"$'\n'"export KUBECONFIG=\"$KUBECONFIG\""
```

Comment:
So the user is going to be told twice to do this `export`?

Reply:
ok, I had a conflicting thought about this... so I'm ok to change it to what make more sense
Having it summarize at the end is nice and evident and convenient for the user.
However, if the script fails, it may not get to the end and it may be useful to have the info available.

Technically it is 3 times because we explicitly put in also in the quickstart for copy&paste convenence.

Reply:
You know about the `trap` builtin command of bash?  It can be used to ensure something happens even if the script crashes.  It is like `try .. finally` in Python.